src/subscribe/OrderMessages.js

```diff
@@ -53,22 +59,44 @@ export default function OrderMessages(client, options = {}) {
             resendStreams.delete(resendMessageStream)
             await resendMessageStream.cancel()
         }
-    }, gapFillTimeout, retryResendAfter)
+    }, gapFillTimeout, retryResendAfter, gapFill ? 5 : 0)
```

Comment:
Introduce a constant for this `5`? Just to avoid magic numbers. Could also add a comment if there is some specific reason why we use this value.

Reply:
Yep agreed.

IMO the problem is actually more that the function takes ~6~ 7 positional arguments.

Loose rule I try to follow is no more than 2±1 positional arguments, otherwise switch to named arguments, particularly when you have a bunch of different "options". Exceptions to rule can be made when the positions make intuitive sense or for trailing object/callback argument.
 
In a future PR I'm going to switch the ordering util to look more like this:

```js
const orderingUtil = new OrderingUtil({
  streamId,
  streamPartition,
  propagationTimeout,
  resendTimeout,
  maxGapRequests: gapFill ? 5 : 0,
})

orderingUtil
    .on('message', inOrderHandler)
    .on('gap', gapHandler)
```

> Could also add a comment if there is some specific reason why we use this value

Yeah no particular reason just that if `resendTimeout` defaults to 5s then the `maxGapRequests = 10` default means stream is blocked for 50s waiting for a single message, and that seems like an awfully long time to me.

Reply:
Added `maxGapRequests` as a client config option in 6e83a422.